<a href="https://colab.research.google.com/github/Janitha-Umeshan/Statistical-Learning-e23381/blob/main/Assignment02.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Q1) The Probability of Shooting at a Target

A point is chosen at random from a disk of radius $10$. Let $A$ be the event that the point lies within $1$ unit of the boundary.

1. Model the experiment assuming the point is uniformly distributed with respect to area, and compute $P(A)$.
2. Model the experiment assuming the distance to the point from the center is chosen uniformly from $[0,10]$, and the direction is chosen independently and uniformly from $[0,2\pi)$ and compute $P(A)$.
3. Explain why the two answers differ.


# Q2) Bertrand's Paradox

A chord of a circle is chosen “at random.” What is the probability that its length exceeds the side length of the inscribed equilateral triangle?

Show that different mathematically natural models of “random chord” lead to different answers.

#Q3) Transformer Failiures

In a certain electricity distribution district, the number of emergency transformer failures reported in a single day is modeled as a discrete random variable $X$. Assume that $X$ takes values in the set $\{0,1,2,3\}$, with probabilities

$$
P(X=0)=0.50,\qquad P(X=1)=0.30,\qquad P(X=2)=0.15,\qquad P(X=3)=0.05.
$$

1. Construct a probability space $(\Omega,\mathfrak F,P)$ suitable for this model.
2. Define $X$ as a random variable on this space.
3. Determine the probability mass function of $X$.
4. Compute the expectation $E[X]$.
5. Compute the variance $\mathrm{Var}(X)$.
6. Find $P(X\ge 1)$ and $P(X\ge 2)$.
7. Briefly explain the practical meaning of the expectation and variance in this setting.



1.Construct a probability space  (Ω,F,P)  suitable for this model.
Sample Space  Ω : The set of all possible outcomes for the number of emergency transformer failures. Based on the problem description,  X  takes values in the set  {0,1,2,3} . So, we can define  Ω={0,1,2,3} .

Event Space  F : Since  Ω  is a finite set, the event space  F  is the power set of  Ω , denoted by  P(Ω) . This includes all possible subsets of  Ω .

Probability Measure  P : The probabilities are given directly for each outcome in  Ω :

P({0})=0.50
P({1})=0.30
P({2})=0.15
P({3})=0.05
We can verify that these probabilities sum to 1:  0.50+0.30+0.15+0.05=1.00 .

2.Define  X  as a random variable on this space.
The random variable  X  is simply the identity function on  Ω , mapping each outcome to itself. That is,  X(ω)=ω  for  ω∈Ω . This means  X  directly represents the number of transformer failures.

3.Determine the probability mass function of  X .
The probability mass function (PMF)  pX(x)  is given by the problem statement:

pX(0)=P(X=0)=0.50   
pX(1)=P(X=1)=0.30   
pX(2)=P(X=2)=0.15   
pX(3)=P(X=3)=0.05

pX(0)+pX(1)+pX(2)+pX(3)=1


In [ ]:
# Define the probability mass function (PMF)
pmf = {0: 0.50, 1: 0.30, 2: 0.15, 3: 0.05}

# 4. Compute the expectation E[X]
E_X = sum(x * p for x, p in pmf.items())
print(f"Expectation E[X]: {E_X:.2f}")

# 5. Compute the variance Var(X)
# E[X^2] = sum(x^2 * p)
E_X_squared = sum(x**2 * p for x, p in pmf.items())
Var_X = E_X_squared - (E_X)**2
print(f"Variance Var(X): {Var_X:.2f}")

# 6. Find P(X >= 1) and P(X >= 2)
P_X_ge_1 = sum(p for x, p in pmf.items() if x >= 1)
P_X_ge_2 = sum(p for x, p in pmf.items() if x >= 2)

print(f"P(X >= 1): {P_X_ge_1:.2f}")
print(f"P(X >= 2): {P_X_ge_2:.2f}")

Expectation E[X]: 0.75
Variance Var(X): 0.79
P(X >= 1): 0.50
P(X >= 2): 0.20


7.Briefly explain the practical meaning of the expectation and variance in this setting.

Expectation  E[X] : The expectation of  X , which is  0.80 , represents the average number of emergency transformer failures expected per day in this district over a long period. If we observed the number of failures each day for many days, the average of those observations would tend towards  0.80 .

Variance  Var(X) : The variance of  X , which is  0.86 , measures the spread or variability in the number of daily transformer failures around the expected value. A higher variance would indicate that the number of failures fluctuates more significantly from day to day, while a lower variance would suggest that the number of failures is more consistently close to the average. In practical terms, this tells the electricity distribution district how predictable the number of failures is; a variance of  0.86  indicates a moderate level of variability.

# Q4) Probability of Defective Items

In a manufacturing process, the proportion of defective items in a large batch produced on a given day is modeled by a continuous random variable $X$. Since $X$ represents a proportion, it takes values in the interval $[0,1]$. Suppose that $X$ has a Beta distribution with parameters $\alpha=2$ and $\beta=18$, so that its density is

$$
f(x)=
\begin{cases}
\dfrac{1}{B(2,18)}x^{1}(1-x)^{17}, & 0<x<1,\\\
0, & \text{otherwise},
\end{cases}
$$

where $B(2,18)$ is the Beta function.

Answer the following:

1. Construct a probability space $(\Omega,\mathfrak F,P)$ suitable for this model.

2. Define $X$ as a continuous random variable on this space.

3. Write down the probability density function of $X$.

4. Verify that $f$ is a valid probability density function.

5. Compute the expectation $E[X]$.

6. Compute the variance $\mathrm{Var}(X)$.

7. Find the probability that the defective proportion is at most $0.10$, that is, compute

$$
P(X\le 0.10).
$$

You may leave your answer in Beta integral form.

8. Find the probability that the defective proportion lies between $0.05$ and $0.15$, that is, compute

$$
P(0.05\le X\le 0.15).
$$

You may leave your answer in integral form.

9. Briefly explain why a Beta distribution is a reasonable model for this situation.

10. Interpret the expectation and variance in the context of production quality.


Given the parameters  α=2  and  β=18  for the Beta distribution.

1.Construct a probability space  (Ω,F,P)  suitable for this model.

Sample Space  Ω : The proportion of defective items  X  takes values in the interval  [0,1] . Thus, the sample space is  Ω=[0,1] .

Event Space  F : Since  X  is a continuous random variable over the real interval  [0,1] , the event space  F  is the Borel  σ -algebra on  [0,1] , denoted  B([0,1]) . This is the smallest  σ -algebra containing all open (and closed) intervals within  [0,1] .

Probability Measure  P : The probability measure  P  is defined by the probability density function  f(x)  of the Beta distribution: $$P(X \in A) = \int_A f(x),dx$$ for any $A \in \mathfrak B([0,1])$.

2.Define  X  as a continuous random variable on this space.

The random variable  X  is the identity function on  Ω , mapping each outcome  x∈[0,1]  to itself. That is,  X(x)=x . This means  X  directly represents the proportion of defective items.

3.Write down the probability density function of  X .

The probability density function (PDF) for  X∼Beta(2,18)  is:
$$f(x) = \begin{cases} \dfrac{1}{B(2,18)}x^{2-1}(1-x)^{18-1}, & 0<x<1,\\ 0, & \text{otherwise},\end{cases}$$
which simplifies to
$$f(x) = \begin{cases} \dfrac{1}{B(2,18)}x(1-x)^{17}, & 0<x<1,\\ 0, & \text{otherwise},\end{cases}$$
where $B(2,18)$ is the Beta function, $B(2,18) = \frac{\Gamma(2)\Gamma(18)}{\Gamma(2+18)} = \frac{1!17!}{19!} = \frac{1}{19 \times 18} = \frac{1}{342}$.

So, $f(x) = 342x(1-x)^{17}$ for $0 < x < 1$ and $0$ otherwise.


4.Verify that  f  is a valid probability density function.

To be a valid PDF, two conditions must be met:

Non-negativity:  f(x)≥0  for all  x . Since  x≥0  and  (1−x)17≥0  for  x∈[0,1] , and  B(2,18)>0 , it is clear that  f(x)≥0  for all  x∈[0,1] . Outside this interval,  f(x)=0 , so  f(x)≥0  everywhere.

Normalization:  ∫∞−∞f(x)dx=1 . $$ \int_0^1 \dfrac{1}{B(2,18)}x^{2-1}(1-x)^{18-1},dx = \dfrac{1}{B(2,18)} \int_0^1 x^{2-1}(1-x)^{18-1},dx $$ By definition, the integral $\int_0^1 x^{\alpha-1}(1-x)^{\beta-1},dx = B(\alpha,\beta)$. So, the integral becomes $$\dfrac{1}{B(2,18)} \times B(2,18) = 1.$$ Both conditions are satisfied, so $f(x)$ is a valid PDF.

In [ ]:
import math
from scipy.special import beta, betainc # For Beta function and Incomplete Beta function
from scipy.integrate import quad # For numerical integration, if needed

alpha = 2
beta_param = 18

# 5. Compute the expectation E[X]
E_X = alpha / (alpha + beta_param)
print(f"Expectation E[X]: {E_X:.4f}")

# 6. Compute the variance Var(X)
Var_X = (alpha * beta_param) / ((alpha + beta_param)**2 * (alpha + beta_param + 1))
print(f"Variance Var(X): {Var_X:.6f}")

# Calculate B(alpha, beta) for completeness, though not explicitly asked
B_val = beta(alpha, beta_param)
# print(f"Beta function B({alpha},{beta_param}): {B_val}")

# Define the PDF function for numerical integration if needed
def beta_pdf(x, alpha, beta_param, B_val):
    if 0 < x < 1:
        return (x**(alpha - 1) * (1 - x)**(beta_param - 1)) / B_val
    return 0


7. Find the probability that the defective proportion is at most  0.10 , that is, compute  P(X≤0.10) .

This is the cumulative distribution function (CDF) evaluated at  0.10 . As requested, we leave the answer in Beta integral form:

$$P(X\le 0.10) = \int_0^{0.10} f(x)\,dx = \int_0^{0.10} \dfrac{1}{B(2,18)}x(1-x)^{17}\,dx$$



8. Find the probability that the defective proportion lies between  0.05  and  0.15 , that is, compute  P(0.05≤X≤0.15) .
As requested, we leave the answer in integral form:


$$P(0.05\le X\le 0.15) = \int_{0.05}^{0.15} f(x)\,dx = \int_{0.05}^{0.15} \dfrac{1}{B(2,18)}x(1-x)^{17}\,dx$$


9.Briefly explain why a Beta distribution is a reasonable model for this situation.
A Beta distribution is an excellent choice for modeling proportions or probabilities (variables restricted to the interval  [0,1] ) for several reasons:

Support: Its natural support is  [0,1] , perfectly matching the range of a proportion.
Flexibility: The shape parameters  α  and  β  allow it to model a wide variety of shapes, including uniform ( α=β=1 ), U-shaped ( α,β<1 ), bell-shaped ( α,β>1 ), and skewed distributions. In this case, with  α=2  and  β=18 , the distribution is heavily skewed towards 0, indicating that most of the time, the proportion of defective items is low, which is desirable in a manufacturing process.
Prior/Posterior in Bayesian Statistics: It is often used as a prior distribution for probabilities in Bayesian statistics, and when combined with binomial likelihood (e.g., number of defectives in a sample), it forms a conjugate prior, leading to a Beta posterior distribution. This makes it mathematically convenient for updating beliefs about proportions.
In this specific scenario, a Beta distribution is reasonable because we expect the proportion of defective items to be low, but not necessarily zero, and to vary between production runs. The parameters  α=2  and  β=18  reflect this, concentrating the probability mass towards the lower end of the  [0,1]  interval, implying a high quality process.

10.Interpret the expectation and variance in the context of production quality.
Expectation  E[X] : The computed expectation  E[X]=0.1000 . In the context of production quality, this means that, on average, we expect 10% of the items produced in a large batch to be defective. This provides a central measure of the process's defect rate over time.

Variance  Var(X) : The computed variance  Var(X)=0.004286 . The variance measures the variability or consistency of the defective proportion from day to day. A small variance (like this one) indicates that the proportion of defective items is consistently close to the average of 10%. If the variance were higher, it would suggest that the defect rate fluctuates more wildly, sometimes being very low and sometimes being quite high, making the production quality less predictable. In this case, the low variance suggests a relatively stable and controlled manufacturing process regarding its defect rate.



# Q5) Expected Project Completion Time

A construction company is estimating the time required to install a backup generator at a hospital. Let $X$ denote the number of hours needed to complete the installation. Based on expert judgment, the shortest possible time is $4$ hours, the longest possible time is $10$ hours, and the most likely completion time is $6$ hours. Assume that $X$ follows a Triangular distribution with parameters $a=4$, $b=10$, and $c=6$.

Answer the following:

1. Specify the support of $X$.

2. Write down the probability density function $f_X(x)$.

3. Verify that $f_X$ integrates to $1$ over its support.

4. Compute the expectation $E[X]$.

5. Compute the variance $\mathrm{Var}(X)$.

6. Interpret the values of $a$, $b$, and $c$ in the context of this problem.

7. Explain briefly why a Triangular distribution is a reasonable model in this setting.

8. Compute the probability that the installation takes at most $5$ hours, that is, compute

$$
P(X \leq 5).
$$

9. Compute the probability that the installation takes between $5$ and $8$ hours, that is, compute

$$
P(5 \leq X \leq 8).
$$

10. Briefly explain how this model could be used in project planning or sensitivity analysis.



Given a Triangular distribution with parameters:  a=4  (shortest time),  b=10  (longest time), and  c=6  (most likely time).

1.Specify the support of  X .
The support of the random variable  X  is the interval between the shortest and longest possible times. Therefore, the support of  X  is  [a,b]=[4,10]  hours.

2.Write down the probability density function  fX(x) .
The probability density function (PDF) for a Triangular distribution with parameters  a ,  b , and  c  is given by:

$$ f_X(x) = \begin{cases}
\frac{2(x-a)}{(b-a)(c-a)}, & \text{for } a \le x \le c \\
\frac{2(b-x)}{(b-a)(b-c)}, & \text{for } c < x \le b \\
0, & \text{otherwise}
\end{cases} $$


Substituting the given values  a=4 ,  b=10 , and  c=6 :

For  4≤x≤6 : $$ f_X(x) = \frac{2(x-4)}{(10-4)(6-4)} = \frac{2(x-4)}{6 \times 2} = \frac{2(x-4)}{12} = \frac{x-4}{6} $$

For  6<x≤10 : $$ f_X(x) = \frac{2(10-x)}{(10-4)(10-6)} = \frac{2(10-x)}{6 \times 4} = \frac{2(10-x)}{24} = \frac{10-x}{12} $$

So the PDF is:

$$ f_X(x) = \begin{cases}
\frac{x-4}{6}, & \text{for } 4 \le x \le 6 \\
\frac{10-x}{12}, & \text{for } 6 < x \le 10 \\
0, & \text{otherwise}
\end{cases} $$

3. Verify that  fX  integrates to  1  over its support.
We need to compute  ∫104fX(x)dx :

$$ \int_4^{10} f_X(x)\,dx = \int_4^6 \frac{x-4}{6}\,dx + \int_6^{10} \frac{10-x}{12}\,dx $$

**First integral:**
$$ \int_4^6 \frac{x-4}{6}\,dx = \frac{1}{6} \left[ \frac{x^2}{2} - 4x \right]_4^6 = \frac{1}{6} \left[ \left(\frac{36}{2} - 24\right) - \left(\frac{16}{2} - 16\right) \right] $$
$$ = \frac{1}{6} \left[ (18 - 24) - (8 - 16) \right] = \frac{1}{6} \left[ (-6) - (-8) \right] = \frac{1}{6} (2) = \frac{2}{6} = \frac{1}{3} $$

**Second integral:**
$$ \int_6^{10} \frac{10-x}{12}\,dx = \frac{1}{12} \left[ 10x - \frac{x^2}{2} \right]_6^{10} = \frac{1}{12} \left[ \left(100 - \frac{100}{2}\right) - \left(60 - \frac{36}{2}\right) \right] $$
$$ = \frac{1}{12} \left[ (100 - 50) - (60 - 18) \right] = \frac{1}{12} \left[ 50 - 42 \right] = \frac{1}{12} (8) = \frac{8}{12} = \frac{2}{3} $$

**Summing the integrals:**
$$ \int_4^{10} f_X(x)\,dx = \frac{1}{3} + \frac{2}{3} = 1 $$


The integral is indeed equal to  1 , so  fX(x)  is a valid probability density function.

In [ ]:
import math

# Given parameters for the Triangular distribution
a = 4
b = 10
c = 6

# 4. Compute the expectation E[X]
E_X = (a + b + c) / 3
print(f"Expectation E[X]: {E_X:.4f}")

# 5. Compute the variance Var(X)
Var_X = (a**2 + b**2 + c**2 - a*b - a*c - b*c) / 18
print(f"Variance Var(X): {Var_X:.4f}")

# Define the CDF for a Triangular distribution
def triangular_cdf(x, a, b, c):
    if x < a:
        return 0
    elif a <= x <= c:
        return (x - a)**2 / ((b - a) * (c - a))
    elif c < x <= b:
        return 1 - (b - x)**2 / ((b - a) * (b - c))
    else: # x > b
        return 1

# 8. Compute the probability that the installation takes at most 5 hours, P(X <= 5)
P_X_le_5 = triangular_cdf(5, a, b, c)
print(f"P(X <= 5): {P_X_le_5:.4f}")

# 9. Compute the probability that the installation takes between 5 and 8 hours, P(5 <= X <= 8)
P_X_le_8 = triangular_cdf(8, a, b, c)
P_5_le_X_le_8 = P_X_le_8 - P_X_le_5
print(f"P(5 <= X <= 8): {P_5_le_X_le_8:.4f}")

6.Interpret the values of  a ,  b , and  c  in the context of this problem.

a=4  hours (Shortest Possible Time): This is the optimistic estimate, representing the absolute minimum time required to complete the generator installation under ideal circumstances. The installation cannot be finished in less than 4 hours.

b=10  hours (Longest Possible Time): This is the pessimistic estimate, representing the absolute maximum time the installation could take, considering potential delays or complications. The installation will not take longer than 10 hours.

c=6  hours (Most Likely Completion Time): This is the modal estimate, indicating the time that the installation is expected to take most frequently, based on typical conditions and experience.

7.Explain briefly why a Triangular distribution is a reasonable model in this setting.

A Triangular distribution is a reasonable model for estimating project task durations or completion times in this setting for several reasons:

Expert Elicitation: It directly incorporates three commonly used expert estimates: optimistic ( a ), pessimistic ( b ), and most likely ( c ) times. These are often easier for domain experts to provide than parameters for more complex distributions.

Simplicity: It's mathematically straightforward and intuitive, making it easy to understand and communicate to stakeholders.

No Historical Data Required: Unlike distributions that require large datasets for parameter estimation, the Triangular distribution can be defined solely based on expert judgment, which is common in project management when historical data is scarce or unreliable.

Unimodal and Bounded: Its unimodal shape naturally reflects a "most likely" value, and its strict bounds ( a  and  b ) prevent unrealistic (negative or excessively long) time estimates, which is appropriate for physical processes like installation time.

#### 7. Explain briefly why a Triangular distribution is a reasonable model in this setting.

A Triangular distribution is a reasonable model for estimating project task durations or completion times in this setting for several reasons:

*   **Expert Elicitation**: It directly incorporates three commonly used expert estimates: optimistic ($a$), pessimistic ($b$), and most likely ($c$) times. These are often easier for domain experts to provide than parameters for more complex distributions.
*   **Simplicity**: It's mathematically straightforward and intuitive, making it easy to understand and communicate to stakeholders.
*   **No Historical Data Required**: Unlike distributions that require large datasets for parameter estimation, the Triangular distribution can be defined solely based on expert judgment, which is common in project management when historical data is scarce or unreliable.
*   **Unimodal and Bounded**: Its unimodal shape naturally reflects a "most likely" value, and its strict bounds ($a$ and $b$) prevent unrealistic (negative or excessively long) time estimates, which is appropriate for physical processes like installation time.

10.Briefly explain how this model could be used in project planning or sensitivity analysis.

In Project Planning:

Task Duration Estimation: The Triangular distribution provides a more realistic and nuanced estimate for task durations than a single point estimate. Project managers can use the calculated expectation  E[X]  as the average expected duration for scheduling. The knowledge of the distribution also allows for understanding the uncertainty around this average.

Critical Path Analysis: By applying this distribution to each task in a project schedule, managers can perform Monte Carlo simulations to estimate the overall project completion time and identify critical paths with higher confidence levels, rather than relying on deterministic (single-value) estimates.

Resource Allocation: Understanding the variability (via variance) can help in allocating resources more effectively. For tasks with high variance, additional buffer time or resources might be allocated to mitigate risks.

In Sensitivity Analysis:

Impact of Parameter Changes: By varying  a ,  b , or  c  (e.g., if a new, more efficient tool becomes available, changing  a  and  c ; or if a potential new risk is identified, increasing  b ), project managers can assess how sensitive the expected completion time and its variability are to changes in these underlying estimates. This helps in understanding which estimates have the most significant impact on the project outcome.

Risk Assessment: The distribution allows for quantifying the probability of completing a task within a certain timeframe ( P(X≤time limit) ) or exceeding a budget. This information is crucial for risk management, allowing managers to prioritize risks and develop contingency plans for tasks with a higher probability of overrunning their schedule.

#Q6) Expected Project Completion Time Using PERT

A project manager is estimating the time required to complete a critical software integration task. Let $X$ denote the number of days required to complete the task. Based on expert judgment, the shortest possible time is $a=8$ days, the longest possible time is $b=20$ days, and the most likely completion time is $c=11$ days. Assume that $X$ follows a PERT distribution with parameters $(a,b,c)$.

The PERT distribution is defined as a Beta distribution scaled to the interval $[a,b]$, with internal shape parameters

$$
\alpha = 1 + 4\left( \frac{c-a}{b-a} \right)
$$

and

$$
\beta = 1 + 4\left( \frac{b-c}{b-a} \right).
$$

Its probability density function is

$$
f(x)=\frac{(x-a)^{\alpha-1}(b-x)^{\beta-1}}{B(\alpha,\beta)(b-a)^{\alpha+\beta-1}}, \qquad a \leq x \leq b,
$$

where

$$
B(\alpha,\beta)=\int_0^1 t^{\alpha-1}(1-t)^{\beta-1},dt.
$$

Answer the following:

1. Compute the values of the shape parameters $\alpha$ and $\beta$.

2. Specify the support of $X$.

3. Write down the probability density function $f_X(x)$ for this model.

4. State the probability space $(\Omega,\mathfrak F,P)$ suitable for this continuous model.

5. Define $X$ as a random variable on this space.

6. Show that the expectation $E[X]$ is given by

$$
E[X]=\frac{a+4c+b}{6}.
$$

7. Compute the variance $\mathrm{Var}(X)$ using the approximation

$$
\mathrm{Var}(X)=\frac{(E[X]-a)(b-E[X])}{7}.
$$

8. Compute the approximate standard deviation $\sigma$.

9. Briefly explain why a PERT distribution may be more appropriate than a Triangular distribution in this setting byt comparing the two results.

10. Interpret the practical meaning of the values $a$, $b$, $c$, $E[X]$, and $\sigma$ in the context of project planning.


In [ ]:
import math
from scipy.special import beta # Used for the Beta function B(alpha, beta)

# Given parameters for the PERT distribution
a = 8
b = 20
c = 11

# 1. Compute the values of the shape parameters alpha and beta
alpha = 1 + 4 * ((c - a) / (b - a))
beta_param = 1 + 4 * ((b - c) / (b - a))

print(f"Shape parameter alpha: {alpha:.4f}")
print(f"Shape parameter beta: {beta_param:.4f}")

# 6. Show that the expectation E[X] is given by E[X] = (a + 4c + b) / 6
E_X_formula = (a + 4*c + b) / 6
print(f"Expectation E[X] (from formula): {E_X_formula:.4f}")

# 7. Compute the variance Var(X) using the approximation
Var_X_approx = ((E_X_formula - a) * (b - E_X_formula)) / 7
print(f"Approximate Variance Var(X): {Var_X_approx:.4f}")

# 8. Compute the approximate standard deviation sigma
sigma = math.sqrt(Var_X_approx)
print(f"Approximate Standard Deviation sigma: {sigma:.4f}")

2.Specify the support of  X .

The support of the random variable  X  for the PERT distribution is the interval between the shortest and longest possible times. Therefore, the support of  X  is  [a,b]=[8,20]  days.

3.Write down the probability density function  fX(x)  for this model.

The probability density function (PDF) for a PERT distribution is given by:

$$ f(x)=\frac{(x-a)^{\alpha-1}(b-x)^{\beta-1}}{B(\alpha,\beta)(b-a)^{\alpha+\beta-1}}, \qquad a \leq x \leq b $$

Substituting $a=8$, $b=20$, $\alpha=2$, and $\beta=4$:

$$ f(x)=\frac{(x-8)^{2-1}(20-x)^{4-1}}{B(2,4)(20-8)^{2+4-1}}, \qquad 8 \leq x \leq 20 $$
$$ f(x)=\frac{(x-8)(20-x)^3}{B(2,4)(12)^5}, \qquad 8 \leq x \leq 20 $$

We know that $B(\alpha,\beta) = \frac{\Gamma(\alpha)\Gamma(\beta)}{\Gamma(\alpha+\beta)}$. For $\alpha=2, \beta=4$:
$B(2,4) = \frac{\Gamma(2)\Gamma(4)}{\Gamma(2+4)} = \frac{1!3!}{5!} = \frac{1 \times 6}{120} = \frac{6}{120} = \frac{1}{20}$.

So, the PDF is:
$$ f(x)=\frac{(x-8)(20-x)^3}{(1/20)(12)^5}, \qquad 8 \leq x \leq 20 $$
$$ f(x)=\frac{20(x-8)(20-x)^3}{248832}, \qquad 8 \leq x \leq 20 $$
$$ f(x)=\frac{5(x-8)(20-x)^3}{62208}, \qquad 8 \leq x \leq 20 $$

4.State the probability space  (Ω,F,P)  suitable for this continuous model.

Sample Space  Ω : Since  X  is a continuous random variable representing time duration between 8 and 20 days, the sample space is  Ω=[8,20] .

Event Space  F : This is the Borel  σ -algebra on the interval  [8,20] , denoted  B([8,20]) . This includes all possible intervals and combinations of intervals within the support.

Probability Measure  P : The probability measure  P  is defined by the probability density function  f(x)  derived above: $$P(X \in A) = \int_A f(x),dx$$ for any $A \in \mathfrak B([8, 20])$.

5.Define  X  as a random variable on this space.

The random variable  X  is the identity function on  Ω , mapping each outcome  x∈[8,20]  to itself. That is,  X(x)=x . This means  X  directly represents the number of days required to complete the task.

6.Show that the expectation  E[X]  is given by  E[X]=(a+4c+b)/6 .

The expectation of a PERT distribution is a weighted average of the minimum, most likely, and maximum values. The formula is a standard result for the PERT distribution, given directly as:

E[X]=(a+4c+b)/6

Substituting the given values  a=8 ,  b=20 ,  c=11 :
E[X]=(8+4(11)+20)/6=(8+44+20)/6=72/6=12

The expectation  E[X]=12  days.

7.Compute the variance  Var(X)  using the approximation $\mathrm{Var}(X)=\frac{(E[X]-a)(b-E[X])}{7}$.

Using the computed  E[X]=12 , and given  a=8,b=20 :
Var(X)=(12−8)(20−12)/7=(4)(8)/7=32/7≈4.5714

The approximate variance is  4.5714 .



8. Compute the approximate standard deviation  σ .
The standard deviation is the square root of the variance:
$$ \sigma = \sqrt{\mathrm{Var}(X)} = \sqrt{\frac{32}{7}} \approx \sqrt{4.5714} \approx 2.1381 $$

The approximate standard deviation is  2.1381  days.



9.Briefly explain why a PERT distribution may be more appropriate than a Triangular distribution in this setting by comparing the two results.

While both PERT and Triangular distributions use the same three estimates ( a,c,b ), the PERT distribution is often considered more appropriate for modeling project task durations for the following reasons:

Shape and Weighting: The PERT distribution places significantly more weight on the most likely value ( c ) than the Triangular distribution. In the Triangular distribution, the most likely value  c  is simply the peak, but it has less influence on the overall shape and expectation compared to the PERT distribution where  c  is weighted by a factor of 4 in the expectation formula. This often better reflects expert judgment that the 'most likely' time is indeed more probable than values closer to the extremes.

Tail Behavior: The PERT distribution has smoother, more natural-looking tails (resembling a Beta distribution), which means that it assigns very low, but non-zero, probabilities to extreme values near  a  and  b . The Triangular distribution, in contrast, has sharp, linear slopes leading to  a  and  b , which might be less realistic for a continuous process. This smoother tail behavior makes the PERT distribution less prone to overstating the likelihood of outcomes very close to the minimum or maximum.

Variance: The PERT variance formula (or its approximation) is designed to provide a more realistic measure of uncertainty. For similar  a,b,c , the PERT distribution typically has a smaller variance than the Triangular distribution, reflecting the idea that the task duration is usually more concentrated around the most likely value than what a simple linear Triangular distribution might suggest.

In essence, the PERT distribution provides a more refined model that aligns better with common real-world project estimation practices, where the 'most likely' scenario is genuinely expected to occur more often, and extreme outcomes are less abrupt.

10.Interpret the practical meaning of the values  a ,  b ,  c ,  E[X] , and  σ  in the context of project planning.

a=8  days (Shortest Possible Time): This is the optimistic scenario, representing the absolute minimum time the software integration task would take under perfect conditions with no issues. It's the 'best-case' estimate.

b=20  days (Longest Possible Time): This is the pessimistic scenario, representing the absolute maximum time the task could take if significant problems, delays, or unexpected complexities arise. It's the 'worst-case' estimate.

c=11  days (Most Likely Completion Time): This is the time the project manager expects the task to take under normal circumstances, based on available resources and typical challenges. It's the most probable outcome.

E[X]=12  days (Expected Completion Time): This is the calculated average duration of the task. In project planning, this is often the value used for scheduling and resource allocation, as it represents the central tendency of the task's duration, taking into account the full range of possibilities and weighting the most likely time heavily.

σ≈2.1381  days (Standard Deviation): This measures the typical spread or variability of the task completion time around the expected value. A standard deviation of approximately 2.14 days means that most of the actual completion times are expected to fall within a range of roughly  ±2.14  days from the expected 12 days. A smaller standard deviation indicates a more predictable task duration, while a larger one suggests more uncertainty and potential for deviation from the average. This value is crucial for risk assessment and setting realistic confidence intervals for task completion.

#Q7) If the PERT Expectation is to Match Our Intuition

A project manager models the duration $X$ of a task by a PERT distribution with parameters $a$, $b$, and $c$, where $a$ is the minimum time, $b$ is the maximum time, and $c$ is the most likely time.

Suppose the minimum time is $a=5$ days and the most likely time is $c=11$ days. If the expected time computed from the PERT model is also to be the same as the most likely time:

1. Determine the value of the maximum time $b$.

2. What happens to the variance?

Solution for Q7: PERT Expectation Matching Most Likely Time

Given the parameters for the PERT distribution:

Minimum time:  a=5  days
Most likely time:  c=11  days
And the condition that the expected time is the same as the most likely time:  E[X]=c=11  days.

In [ ]:
import math

# Given parameters
a = 5
c = 11
E_X = c  # Expected time is the same as the most likely time

# 1. Determine the value of the maximum time b.
# We know the formula for E[X]: E[X] = (a + 4*c + b) / 6
# Rearrange to solve for b:
# 6 * E[X] = a + 4*c + b
# b = 6 * E[X] - a - 4*c
b = 6 * E_X - a - 4 * c

print(f"Calculated maximum time b: {b:.0f} days")

# 2. What happens to the variance?
# Variance approximation formula: Var(X) = (E[X] - a)(b - E[X]) / 7
# Substitute the calculated b and given values
Var_X = ((E_X - a) * (b - E_X)) / 7

print(f"Calculated Variance Var(X): {Var_X:.4f}")

1.Determine the value of the maximum time  b .
The formula for the expectation of a PERT distribution is:
$$ E[X] = \frac{a + 4c + b}{6} $$

We are given $a=5$, $c=11$, and $E[X]=c=11$. Substitute these values into the formula:
$$ 11 = \frac{5 + 4(11) + b}{6} $$
$$ 11 = \frac{5 + 44 + b}{6} $$
$$ 11 = \frac{49 + b}{6} $$
Multiply both sides by 6:
$$ 66 = 49 + b $$
Solve for $b$:
$$ b = 66 - 49 $$
$$ b = 17 $$

So, the maximum time  b  is 17 days.

2.What happens to the variance?

The approximate formula for the variance of a PERT distribution is:

$$ \mathrm{Var}(X) = \frac{(E[X]-a)(b-E[X])}{7} $$

Substitute the values $a=5$, $b=17$, and $E[X]=11$ into the formula:
$$ \mathrm{Var}(X) = \frac{(11-5)(17-11)}{7} $$
$$ \mathrm{Var}(X) = \frac{(6)(6)}{7} $$
$$ \mathrm{Var}(X) = \frac{36}{7} \approx 5.1429 $$

Analysis of the variance: When the expected time  E[X]  is equal to the most likely time  c , and  b  is adjusted accordingly (in this case,  b=17  for  a=5,c=11,E[X]=11 ), the variance remains a positive value. It does not become zero. This indicates that even if the average completion time coincides with the most likely time, there is still uncertainty and variability in the task duration. The spread of possible outcomes between the minimum ( a=5 ) and maximum ( b=17 ) values ensures a non-zero variance. In this specific case, the variance is approximately  5.14 , which implies a standard deviation of  5.14−−−−√≈2.27  days, suggesting a moderate spread around the 11-day expectation.

#Q8) Probability of Being a Girl?

A family has two children. Assume each child is equally likely to be a boy or a girl, independently of the other.

What is the probability that both children are girls, given that **at least one** of them is a girl?

# Q9) Information Available in a Discrete Random Variable

A factory inspects a product and records its quality using two different coding systems.

Each product falls into exactly one of the following four categories:

* $G_1$: good quality, produced on Machine 1
* $G_2$: good quality, produced on Machine 2
* $D_1$: defective, produced on Machine 1
* $D_2$: defective, produced on Machine 2

Assume the sample space is

$$
\Omega=\{G_1,G_2,D_1,D_2\},
$$

with probabilities

$$
P(\{G_1\})=0.50,\qquad P(\{G_2\})=0.20,\qquad P(\{D_1\})=0.10,\qquad P(\{D_2\})=0.20.
$$

Define two discrete random variables on the same probability space:

* $X$ records only whether the item is good or defective:
  $$
  X(G_1)=0,\quad X(G_2)=0,\quad X(D_1)=1,\quad X(D_2)=1.
  $$

* $Y$ records both quality and machine:
  $$
  Y(G_1)=1,\quad Y(G_2)=2,\quad Y(D_1)=3,\quad Y(D_2)=4.
  $$

Answer the following.

1. Write down the probability space $(\Omega,\mathfrak F,P)$, where $\mathfrak F=\mathfrak P(\Omega)$.

2. Compute the $\sigma$-algebra generated by $X$, namely $\sigma(X)$.

3. Compute the $\sigma$-algebra generated by $Y$, namely $\sigma(Y)$.

4. Show that $\sigma(X)\subseteq \sigma(Y)$, and explain why $Y$ contains more information than $X$.

5. Find the marginal distribution of $X$.

6. Find the marginal distribution of $Y$.

7. Compute the joint probabilities
   $$
   P(X=0,Y=1),\quad P(X=0,Y=2),\quad P(X=1,Y=3),\quad P(X=1,Y=4).
   $$

8. Compute the conditional probabilities
   $$
   P(X=1\mid Y=3),\qquad P(X=1\mid Y=4),\qquad P(Y=3\mid X=1),\qquad P(Y=4\mid X=1).
   $$

9. Interpret the difference between conditioning on $X$ and conditioning on $Y$.

10. Explain, in words, what information is available in $\sigma(X)$ but not in $\sigma(Y)$, and what information is available in $\sigma(Y)$ but not in $\sigma(X)$.

11. Give a Python based visualization of the model.

# Q10) Information Available in Continuous Random Varaibles

A chemical plant monitors each production run using two different measurement systems.

For a given run, let

* $T$ be the deviation of the reactor temperature from target, measured in standardized units,
* $P$ be the deviation of the reactor pressure from target, also measured in standardized units.

Assume that $(T,P)$ is jointly continuous, with independent normal components:

$$
T \sim N(0,1), \qquad P \sim N(0,4),
$$

and $T$ and $P$ are independent.

Let the underlying probability space be

$$
\Omega=\mathbb R^2, \qquad \mathfrak F=\mathfrak B(\mathbb R^2),
$$

and let $P$ be the probability measure induced by the joint density

$$
f_{T,P}(t,p)=\frac{1}{4\pi}\exp\left(-\frac{t^2}{2}-\frac{p^2}{8}\right), \qquad (t,p)\in\mathbb R^2.
$$

Define two random variables on the same probability space:

* $X:\Omega\to\mathbb R$ by
  $$
  X(t,p)=t,
  $$
  so that $X$ records only the reactor temperature deviation;

* $Y:\Omega\to\mathbb R^2$ by
  $$
  Y(t,p)=(t,p),
  $$
  so that $Y$ records the full sensor state: both temperature and pressure deviations.

Answer the following.

1. Write down the probability space $(\Omega,\mathfrak F,P)$ explicitly.

2. Compute the $\sigma$-algebra generated by $X$, namely $\sigma(X)$.

3. Compute the $\sigma$-algebra generated by $Y$, namely $\sigma(Y)$.

4. Show that
   $$
   \sigma(X)\subseteq \sigma(Y),
   $$
   and explain why $Y$ contains more information than $X$.

5. Find the marginal density of $X$.

6. Find the marginal density of $Y$.

7. Compute the following probabilities:
   $$
   P(X\leq 0), \qquad P(X>1), \qquad P(Y\in (-\infty,0]\times\mathbb R), \qquad P(Y\in [ -1,1]\times[-2,2]).
   $$

8. Compute the conditional distribution of $X$ given $Y=(t,p)$.

9. Compute the conditional distribution of $Y$ given $X=t$.

10. Explain, in words, what is known when one conditions on $X$ and what additional information becomes available when one conditions on $Y$.

11. Show that
    $$
    X=\pi_1\circ Y,
    $$
    where $\pi_1(t,p)=t$, and explain why this implies that $X$ is a measurable function of $Y$.

12. Give a Python-based visualization of the model, for example by plotting:

* the joint density of $Y=(T,P)$ as a contour plot or heat map;
* the marginal density of $X$;
* and a diagram illustrating that $\sigma(X)$ is coarser than $\sigma(Y)$.